kong/api/routes/certificates.lua

```diff
@@ -98,13 +101,129 @@ return {
+      end
+
+      local snis = nil
+      if type(self.params.snis) == "string" then
+        snis = utils.split(self.params.snis, ",")
+      end
+
+      self.params.snis = nil
+
+      local seen = {} -- check for duplicate snis in the request
+      local skip = {} -- avoid db insert if sni is already present in db
+
+      -- if snis field present
+      -- 1. no duplicate snis should be present in the request
+      -- 2. check if any sni in the request is using a cert 
+      --    other than the one being updated
+
+      if snis then
+        for _, sni in ipairs(snis) do
+          if seen[sni] then
+            return helpers.responses.send_HTTP_CONFLICT(
+              "duplicate requested sni name " .. sni
+            )
+          end
+
+          local sni_in_db, err = dao_factory.ssl_servers_names:find({
+            name = sni,
+          })
+          if err then
+            return helpers.yield_error(err)
+          end
+
+          if sni_in_db then
+            if sni_in_db.ssl_certificate_id ~= ssl_cert.id then
+              return helpers.responses.send_HTTP_CONFLICT(
+                "certificate with id " .. sni_in_db.ssl_certificate_id .. 
+                " in use for entry with name " .. sni
+              )
+            end
+            skip[sni] = true
+          end
+          seen[sni] = true
+        end
+      end
+
+      local old_snis, err = dao_factory.ssl_servers_names:find_all {
+        ssl_certificate_id = ssl_cert.id
+      }
+      if err then
+        return helpers.yield_error(err)
+      end
+
+      -- update certificate if necessary
+      if self.params.key and self.params.cert then
+        local err
+        ssl_cert, err = dao_factory.ssl_certificates:update(self.params,{
+          id = self.params.id})
+        if err then
+          return helpers.yield_error(err)
+        end
+      end
+
+      ssl_cert.snis = setmetatable({}, cjson.empty_array_mt)
+
+      if not snis then
+        for i = 1, #old_snis do
+          table.insert(ssl_cert.snis,  old_snis[i].name)
+        end
+        return helpers.responses.send_HTTP_OK(ssl_cert)
+      end
+
+      -- insert/delete SNIs into db if snis field was present in the request
+        for _, sni in ipairs(snis) do
+          if not skip[sni] then
+            local ssl_server_name = {
+              name                = sni,
+              ssl_certificate_id  = ssl_cert.id,
+            }
+            local _, err = dao_factory.ssl_servers_names:insert(
+              ssl_server_name
+            )
+            if err then
+              return helpers.yield_error(err)
+            end
+          end
+          table.insert(ssl_cert.snis, sni)
+        end
+
+        -- delete snis which should no longer use ssl_cert
+        for i = 1, #old_snis do
+          if not seen[old_snis[i].name] then
+            dao_factory.ssl_servers_names:delete(
+                { name = old_snis[i].name }
```

Comment:
Could we follow the same style as usual for this table argument as well? Thank you!